week0002/daudau/php/daudau.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/*
+	explaination of my code. i got it from myself.
+	Do đề bài là theo hình xoắn ốc, ta nhận thấy nó đưọc chia thành các lớp, mỗi lớp là 1 "vòng tròn". đường thẳng cũng là vòng tròn :D
+	-> ý tưởng là dựa vào tọa độ của điểm trong array, ta cần xác định nó thuộc lớp thứ mấy của hình xoắn ốc và xác định giá trị tại điểm đó.
+	Example data:
+	N = 3; M = 4;
+	-------------
+	1	2	3	4
+
+	10	11	12	5
+
+	9	8	7	6
+	-------------
+*/
+
+//nhập giá trị từ user.
+fwrite(STDOUT, 'Input the number of rows N: ');
+fscanf(STDIN, "%d", $n);
+
+fwrite(STDOUT, 'Input the number of columns M: ');
+fscanf(STDIN, "%d", $m);
+
+$min = $m > $n ? $n : $m;  //tìm min của hàng và cột
+$circle = ceil($min / 2.0);  //xác định số vòng mà xoắn ốc sẽ có, dựa vào giá trị min.
+$length = strlen($n*$m) + 1;  //sử dụng để in giá trị
+$roundFirstValue = array();  //mảng này để lưu giá trị bắt đầu của mỗi vòng. VD vòng thứ 0 là 1, vòng thứ 1 là 11.
+
+$roundFirstValue[0] = 1; //giá trị vòng đầu tiên luôn là 1.
+for ($i = 1; $i < $circle; $i++) {
+	/**
+	* giá trị bắt đầu mỗi vòng = giá trị bắt đầu của vòng trước đó + số số của vòng đó.
+	* vd Vòng 1 là 1, và có số trong vòng đó => giá trị bắt đầu của vòng 2 là 1 + 10.
+	* số lưọng phần tử trong 1 vòng bằng : ($n-($i-1)*2 + $m-($i-1)*2)*2 - 4
+	*/
+	$roundFirstValue[$i] = $roundFirstValue[$i - 1] + ($n - ($i - 1) * 2 + $m - ($i - 1) * 2) * 2 - 4;
```

Comment:
It's a bit messy. I understood that if a rectangle has dimensions  M & N then the number of elements is: `2 * (M + N) - 4`